src/main.rs

```diff
@@ -79,19 +84,19 @@ impl VM {
             Opcode::Br => conditional_branch(self, instruction),
             Opcode::Add => add(self, instruction),
             Opcode::Ld => load(self, instruction),
-            Opcode::St => todo!(),
+            Opcode::St => store(self, instruction),
             Opcode::Jsr => jump_subroutine(self, instruction),
             Opcode::And => and(self, instruction),
             Opcode::Ldr => load_register(self, instruction),
-            Opcode::Str => todo!(),
-            Opcode::Rti => todo!(),
+            Opcode::Str => store_register(self, instruction),
+            Opcode::Rti => Ok(()), // RTI is not implemented
```

Comment:
Should you continue execution if you get an `RTI` opcode, I think halting/ returning an error is better if the program being executed contains any forbidden instructions

Reply:
I agree with Lean here, sometimes returning an error, or even panicking is the correct choice.

Reply:
Sure i will improve this logic in next PR!